tests/tcg/arc64/check_qmach.S

```diff
@@ -0,0 +1,553 @@
+; --------------------------------
+; (   1 + 9223372036854775807    )
+; -------------------------------
+;        0x8000_0000_0000_0000
+;        -9223372036854775808
+  prep_test_case
+  set_reg64 r58, hi=0x7fffffff, lo=0xffffffff
+  set_reg64  r2, hi=0x00000000, lo=0x00000001
+  qmach    r2, r2, r2
+  flg_none_set
+  chk_reg64  r2, hi=0x80000000, lo=0x00000000
+
+; Test case 13
+; reg <- reg, u6 (u6 is extended to 16 bits and then replicated 4 times)
+;   32767     32767     32767     32767
+; (0x7fff,   0x7fff,   0x7fff,   0x7fff) * r7
+; (0b111111, 0b111111, 0b111111, 0b111111)   u6
+;      63,       63,       63,       63
+; (     0x0000_0000_0000_0000    )   r58 (acc)
+; --------------------------------
+; (   2064321 * 4 + 0    )
+; -------------------------------
+;    0x0000_0000_007d_ff04
+;                  8257284
+  prep_test_case
+  set_reg64 r58, hi=0x00000000, lo=0x00000000
+  set_reg64  r7, hi=0x7fff7fff, lo=0x7fff7fff
+  qmach    r6, r7, 63
+  flg_none_set
+  chk_reg64  r6, hi=0x00000000, lo=0x007dff04
+
+; Test case 14
+; reg <- reg, s12 (s12 is sign extended to 16 bits and then replicated 4 times)
+;   32767     32767    -32768    -32768
+; (0x7fff,   0x7fff,   0x8000,   0x8000) * r6
+; (    -1,       -1,       -1,       -1)   s12
+; (     0x0000_0000_0000_0000    )   r58 (acc)
+; --------------------------------
+; (   -32767 * 2 + 32768 * 2    )
+; -------------------------------
+;    0x0000_0000_0000_0002
+  prep_test_case
+  set_reg64 r58, hi=0x00000000, lo=0x00000000
+  set_reg64  r6, hi=0x7fff7fff, lo=0x80008000
+  qmach    r6, r6, -1
+  flg_none_set
+  chk_reg64  r6, hi=0x00000000, lo=0x00000002
+
+; Test case 15
+; reg <- reg, s12 (smallest negative s12)
+;   32767     32767    -32768    -32768
+; (0x7fff,   0x7fff,   0x8000,   0x8000) * r9
+; ( -2048,    -2048,    -2048,    -2048)   s12
+; (     0x0000_0000_0000_0000    )   r58 (acc)
+; --------------------------------
+; (   -67106816 * 2 + 67108864 * 2    )
+; (   -134213632 + 134217728    )
+; -------------------------------
+;    0x0000_0000_0000_1000
+;                     4096
+  prep_test_case
+  set_reg64 r58, hi=0x00000000, lo=0x00000000
+  set_reg64  r9, hi=0x7fff7fff, lo=0x80008000
+  qmach    r9, r9, -2048
+  flg_none_set
+  chk_reg64  r9, hi=0x00000000, lo=0x00001000
+
+; Test case 16
+; reg <- reg, s12 (biggest positive s12)
+;   32767     32767    -32768    -32768
+; (0x7fff,   0x7fff,   0x8000,   0x8000) * r7
+; (  2047,     2047,     2047,     2047)   s12
+; (     0x0000_0000_0000_0000    )   r58 (acc)
+; --------------------------------
+; (   67074049 * 2 + -67076096 * 2    )
+; (   -134213632 + 134217728    )
+; -------------------------------
+;    0xffff_ffff_ffff_f002
+;                    -4096
+  prep_test_case
+  set_reg64 r58, hi=0x00000000, lo=0x00000000
+  set_reg64  r7, hi=0x7fff7fff, lo=0x80008000
+  qmach    r7, r7, 2047
+  flg_none_set
+  chk_reg64  r7, hi=0xffffffff, lo=0xfffff002
+
+
+; Test case 17
```

Comment:
set `r7` to a non-zero value and also check acc haven't changed:

```
mov      r58, 0x123
mov       r7, r58
...
chk_reg64 r7, hi=0x0, lo=0x123
```

you don't need to introduce a macro for checking acc.